drivers/npmx/npmx_shell.c

```diff
@@ -70,8 +73,190 @@ typedef enum {
+		int err = 0;
+		switch (arg_info->type) {
+		case SHELL_ARG_TYPE_INT_VALUE:
+			arg_info->result.ivalue = shell_strtol(argv[i], 0, &err);
+			if (err != 0) {
+				shell_error(shell, "Error: %s has to be an integer.",
+					    arg_info->name);
+				return false;
+			}
+			break;
+		case SHELL_ARG_TYPE_UINT_VALUE:
+		case SHELL_ARG_TYPE_UINT_INDEX:
+			arg_info->result.uvalue = shell_strtoul(argv[i], 0, &err);
+			if (err != 0) {
+				shell_error(shell, "Error: %s has to be a non-negative integer.",
+					    arg_info->name);
+				return false;
+			}
+			break;
+		case SHELL_ARG_TYPE_BOOL_VALUE:
+			const char *str = argv[i];
+			/* shell_strtobool can't be used due to accepting any big value as true. */
+			if ((strcmp(str, "on") == 0) || (strcmp(str, "enable") == 0) ||
+			    (strcmp(str, "true") == 0)) {
+				arg_info->result.bvalue = true;
+				break;
+			}
+			if ((strcmp(str, "off") == 0) || (strcmp(str, "disable") == 0) ||
+			    (strcmp(str, "false") == 0)) {
+				arg_info->result.bvalue = false;
+				break;
+			}
+
+			uint32_t bool_value = shell_strtoul(str, 0, &err);
+			if ((err != 0) || (bool_value > 1)) {
+				shell_error(shell, "Error: invalid %s value.", arg_info->name);
+				return false;
+			}
+
+			arg_info->result.bvalue = bool_value == 1;
+			break;
+		}
+	}
+
+	return true;
+}
+
+static npmx_instance_t *npmx_instance_get(const struct shell *shell)
+{
+	npmx_instance_t *npmx_instance = npmx_driver_instance_get(pmic_dev);
+	if (npmx_instance == NULL) {
+		shell_error(shell, "Error: shell is not initialized.");
+		return NULL;
+	}
+
+	return npmx_instance;
+}
+
+static bool check_instance_index(const struct shell *shell, const char *instance_name,
+				 uint32_t index, uint32_t max_index)
+{
+	if (index >= max_index) {
+		shell_error(shell, "Error: %s instance index is too high: no such instance.",
+			    instance_name);
+		return false;
+	}
+
+	return true;
+}
+
+static const char *unit_str_get(unit_type_t unit_type)
+{
+	switch (unit_type) {
+	case UNIT_TYPE_AMPERAS:
+		return " mA";
+	case UNIT_TYPE_VOLTS:
+		return " mV";
+	case UNIT_TYPE_CELSIUS:
+		return "*C";
+	case UNIT_TYPE_OHMS:
+		return " ohms";
+	case UNIT_TYPE_PCT:
+		return "%%";
+	default:
+		return "";
+	}
+}
+
+static void print_value_int(const struct shell *shell, int value, unit_type_t unit_type)
+{
+	shell_print(shell, "Value: %d%s.", value, unit_str_get(unit_type));
+}
+
+#define print_value(shell, value, unit_type) print_value_int((shell), (int)(value), (unit_type))
+
+static void print_success_int(const struct shell *shell, int value, unit_type_t unit_type)
+{
+	shell_print(shell, "Success: %d%s.", value, unit_str_get(unit_type));
+}
+
+#define print_success(shell, value, unit_type) print_success_int((shell), (int)(value), (unit_type))
+
+static void print_help(const struct shell *shell, int32_t index, const char *str)
+{
+	shell_print(shell, "%d-%s", index, str);
+}
+
 static bool check_error_code(const struct shell *shell, npmx_error_t err_code)
```

Comment:
Same as in other comment, I would keep `check_error_code` as the last global helper function, meaning it would be after `check_pin_configuration_correctness`